configuration/pih/scripts/global/zl.js

```diff
@@ -88,3 +88,74 @@ function setUpExpandableContacts(badPhoneNumberMsg) {
 
 }
 
+function setUpDatepickerStartAndEndDateValidation(badDateInTheFutureMsg,badStartDateGreaterThanEndDateMsg){
+
+    jq(".startDateEndDate").each(function(j, domEl){
+        setUpDatepickers(this);
+    });
+
+    function setUpDatepickers(containerNode) {
+        const startDatepicker = jq(jq(containerNode).find('.startDate'));
+        const endDatepicker = jq(jq(containerNode).find('.endDate'));
+
+        startDatepicker.change(function (e) {
+          let startDate = Date.parse(e.target.value);
+          let endDate = Date.parse(endDatepicker.find('input[type=text]').val());
+
+          if (startDate > Date.now()) {
+              jq(this).find('span').show();
+              jq(this).find('span').text(badDateInTheFutureMsg);
+              setButtonsDisabled(true)
+          } else {
+              jq(this).find('span').hide();
+              jq(this).find('span').text('');
+              setButtonsDisabled(false)
+          }
+
+          if (endDate > Date.now()){
+              endDatepicker.find('span').show();
+              endDatepicker.find('span').text(badDateInTheFutureMsg);
+              setButtonsDisabled(true)
+          } else {
+              endDatepicker.find('span').hide();
+              endDatepicker.find('span').text('');
+              setButtonsDisabled(false)
+          }
```

Comment:
This doesn't depend on `startDatepicker`. Let's get rid of it and just have this check in the `endDatepicker.change` function.

Reply:
I do this because if the user changes the start date, you have to do a check on the end date to display an error if so.